src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +50,633 @@ const ActorInit En_Horse_Normal_InitVars = {
+        }
+
+        SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                             SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A6C6B0(EnHorseNormal* this) {
+    this->type = HORSE_STATIC_CLONEABLE;
+    this->animationIdx = 0;
+    this->unk_21C = 0;
+    this->unk_21E = 0;
+    this->actor.flags |= 0x30;
+    this->actor.speedXZ = 0.0f;
+    this->unk_218 = 0.0f;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+}
+
+void EnHorseNormal_WaitClone(EnHorseNormal* this, GlobalContext* globalCtx) {
+    func_80A6BD7C(this);
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        f32 rand = Math_Rand_ZeroOne();
+
+        if (rand < 0.4f) {
+            this->animationIdx = 0;
+        } else if (rand < 0.8f) {
+            this->animationIdx = 1;
+            this->unk_1E4 |= 0x20;
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_GROAN, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        } else {
+            this->animationIdx = 3;
+            this->unk_1E4 |= 0x20;
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+
+        SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                             SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+
+        this->unk_1E4 &= ~1;
+        this->unk_1E4 &= ~2;
+        this->unk_1E4 &= ~8;
+        this->unk_1E4 &= ~0x10;
+    }
+}
+
+void func_80A6C8E0(EnHorseNormal* this, GlobalContext* globalCtx) {
+    s32 pad;
+    CollisionPoly* sp38;
+    s32 pad2;
+    Vec3f sp28;
+    s32 sp24;
+
+    sp28.x = (Math_Sins(this->actor.shape.rot.y) * 30.0f) + this->actor.posRot.pos.x;
+    sp28.y = this->actor.posRot.pos.y + 60.0f;
+    sp28.z = (Math_Coss(this->actor.shape.rot.y) * 30.0f) + this->actor.posRot.pos.z;
+    this->unk_220 = func_8003C940(&globalCtx->colCtx, &sp38, &sp24, &sp28);
+    this->actor.shape.rot.x = Math_atan2f(this->actor.posRot.pos.y - this->unk_220, 30.0f) * (0x8000 / M_PI);
+}
+
+static EnHorseNormalActionFunc sActionFuncs[] = {
+    EnHorseNormal_CycleAnimations, EnHorseNormal_Wander,     EnHorseNormal_Wait,
+    EnHorseNormal_WaitClone,       EnHorseNormal_FollowPath,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Normal/func_80A6BC00.s")
+void EnHorseNormal_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHorseNormal* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Normal/func_80A6BC48.s")
+    sActionFuncs[this->type](this, globalCtx);
```

Comment:
We normally that the `action`
```suggestion
    sActionFuncs[this->action](this, globalCtx);
```